src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -81,28 +86,189 @@ void VulnerabilityScannerFacade::start(
             m_indexerConnector, m_databaseFeedManager, m_reportSocketClient, m_internalMutex);
 
+        // Queues to process skipped events
+        m_deltaQueue = std::make_shared<Utils::SafeQueue<std::string, RocksDBQueue<std::string>>>(
+            RocksDBQueue<std::string>(DELTA_EVENTS_QUEUE_PATH));
+        m_syncQueue = std::make_shared<Utils::SafeQueue<std::string, RocksDBQueue<std::string>>>(
+            RocksDBQueue<std::string>(SYNCHRONIZATION_EVENTS_QUEUE_PATH));
+
+        // Threads to retry skipped events
+        m_deltaThread = std::thread(
+            [scanOrchestrator, this]()
+            {
+                while (!m_stopRetryThread.load())
+                {
+                    for (int i = 0; i < SLEEP_RETRY_THREADS_SEC; ++i)
+                    {
+                        std::this_thread::sleep_for(std::chrono::seconds(1));
+                        if (m_stopRetryThread.load())
+                        {
+                            break;
+                        }
+                    }
+
+                    while (!m_deltaQueue->empty() && !m_stopRetryThread.load())
+                    {
+                        std::shared_ptr<std::string> returnStr = m_deltaQueue->pop();
+
+                        flatbuffers::Verifier deltaVerifier(reinterpret_cast<const uint8_t*>(returnStr->c_str()),
+                                                            returnStr->size());
+
+                        if (SyscollectorDeltas::VerifyDeltaBuffer(deltaVerifier))
+                        {
+                            const SyscollectorDeltas::Delta* delta =
+                                SyscollectorDeltas::GetDelta(reinterpret_cast<const char*>(returnStr->c_str()));
+                            std::variant<const SyscollectorDeltas::Delta*, const SyscollectorSynchronization::SyncMsg*>
+                                data = delta;
+
+                            try
+                            {
+                                logDebug1(WM_VULNSCAN_LOGTAG,
+                                          "Retry dbsync_packages event: %s package",
+                                          (delta->data_as_dbsync_packages() && delta->data_as_dbsync_packages()->name())
+                                              ? delta->data_as_dbsync_packages()->name()->c_str()
+                                              : "");
+                                scanOrchestrator->run(data);
+                            }
+                            catch (const std::exception& e)
+                            {
+                                logWarn(WM_VULNSCAN_LOGTAG,
+                                        "Discarded dbsync_packages event: %s package. %s",
+                                        (delta->data_as_dbsync_packages() && delta->data_as_dbsync_packages()->name())
+                                            ? delta->data_as_dbsync_packages()->name()->c_str()
+                                            : "",
```

Comment:
This expression is really hard to follow, can we declare it as a const variable before the debug call with a descriptive name?

You could also declare the variable outside the try-catch

Reply:
ditto